packages/block-library/src/search/edit.native.js

```diff
@@ -1,30 +1,75 @@
 /**
  * External dependencies
  */
-import { View, Alert } from 'react-native';
+import { View, Alert, TextInput } from 'react-native';
+import classnames from 'classnames';
 
 /**
  * WordPress dependencies
  */
-import { RichText, BlockControls } from '@wordpress/block-editor';
-import { ToolbarGroup, ToolbarButton } from '@wordpress/components';
+import {
+	RichText,
+	BlockControls,
+	useBlockProps,
+} from '@wordpress/block-editor';
+import { ToolbarGroup, ToolbarButton, Button } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
+import { search } from '@wordpress/icons';
 
 /**
  * Internal dependencies
  */
 import { buttonWithIcon, toggleLabel } from './icons';
-import ButtonPositionDropdown from './button-position-dropdown.native';
+import ButtonPositionDropdown from './button-position-dropdown';
+import styles from './style.scss';
+import richTextStyles from './rich-text.scss';
 
-export default function SearchEdit( { attributes, setAttributes } ) {
-	const { label, showLabel, buttonPosition, buttonUseIcon } = attributes;
+const MIN_BUTTON_WIDTH = 100;
+
+export default function SearchEdit( { attributes, setAttributes, className } ) {
+	const {
+		label,
+		showLabel,
+		buttonPosition,
+		buttonUseIcon,
+		placeholder,
+		buttonText,
+	} = attributes;
 
 	// Temporary. Will be removed when styling is implemented
 	// in a future PR.
 	const alert = ( message ) => {
 		Alert.alert( '', message, [ { text: 'OK' } ], { cancelable: true } );
 	};
 
+	const getBlockClassNames = () => {
+		return classnames(
+			className,
+			'button-inside' === buttonPosition
```

Comment:
I think some of this function could be simplified by creating a map 

Something like 
```
const buttonPostionToClassMap = {
 'button-inside' : 'wp-block-search__button-inside',
 'button-outside': 'wp-block-search__button-outside',
 'no-button': 'wp-block-search__no-button',
 'button-only': 'wp-block-search__button-only',
}
let noButtonClass = null;
if( 'no-button' !== buttonPosition ) {
 noButtonClass = buttonUseIcon ? 'wp-block-search__text-button' : 'wp-block-search__text-button';
}
return classnames(
	className,
	buttonPosition && buttonPostionToClassMap [ buttonPosition ],
	noButtonClass
);
```

Let me know what you think :)

Reply:
Thanks for this @enejb ! This PR is already merged, but I have many more in the pipeline so I'll use this code in the future!